private/bufpkg/bufplugin/bufplugindocker/docker_test.go

```diff
@@ -216,27 +222,93 @@ func newDockerServer(t testing.TB, version string) *dockerServer {
 	mux.HandleFunc("/session", dockerServer.sessionHandler)
 	mux.HandleFunc(versionPrefix+"/build", dockerServer.buildHandler)
 	mux.HandleFunc(versionPrefix+"/images/", dockerServer.imagesHandler)
-	dockerServer.httpServer = httptest.NewServer(h2c.NewHandler(mux, &http2.Server{}))
+	dockerServer.h2Server = &http2.Server{}
+	dockerServer.h2Handler = h2c.NewHandler(mux, dockerServer.h2Server)
+	dockerServer.httpServer = httptest.NewUnstartedServer(dockerServer.h2Handler)
+	dockerServer.httpServer.Start()
 	t.Cleanup(dockerServer.httpServer.Close)
 	return dockerServer
 }
 
 func (d *dockerServer) sessionHandler(w http.ResponseWriter, r *http.Request) {
-	if _, err := io.Copy(io.Discard, r.Body); err != nil {
-		d.t.Error("failed to discard body:", err)
-	}
 	if r.Method != http.MethodPost {
 		http.NotFound(w, r)
 		return
 	}
 	if strings.EqualFold(r.Header.Get("Connection"), "upgrade") && r.ProtoMajor < 2 {
```

Comment:
The Docker client isn't sending a required HTTP2-Settings header so the h2c implementation isn't working out of the box: https://cs.opensource.google/go/x/net/+/1bab6f36:http2/h2c/h2c.go;l=177;drc=1bab6f366d9ec8c152f8735b2e359822ef75e3bd